### PR TITLE
Update CEL documentation to capture type checking errors.

### PR DESCRIPTION
**Pull request by whitneygriffith**
Part of [#47533](https://github.com/istio/istio/issues/47533) 

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [x] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/observability/logs/telemetry-api/index.md [hunk 1]
```diff
@@ -132,6 +132,28 @@ $ kubectl apply -f @samples/open-telemetry/loki/otel.yaml@ -n istio-system
     EOF
     {{< /text >}}
 
+1. Filter health check access logs with CEL expression
+
+    The following configuration displays access logs only when the logs aren't generated by the Amazon-Route53-Health-Check-Service.
```
Comment (dhawton) [inline content/en/docs/tasks/observability/logs/telemetry-api/index.md:137]:
```suggestion
    The following configuration displays access logs only when the logs aren't generated by the Amazon-Route53-Health-Check-Service.
```

File: content/en/docs/tasks/observability/logs/telemetry-api/index.md [hunk 2]
```diff
@@ -132,6 +132,28 @@ $ kubectl apply -f @samples/open-telemetry/loki/otel.yaml@ -n istio-system
...
+    metadata:
+      name: filter-health-check-logging
+    spec:
+      accessLogging:
+      - providers:
+        - name: otel
+        filter:
+          expression: "!has(request.useragent) || !(request.useragent.startsWith("Amazon-Route53-Health-Check-Service"))"
+
+    EOF
```
Comment (dhawton) [inline content/en/docs/tasks/observability/logs/telemetry-api/index.md:154]:
```suggestion
          expression: "!has(request.useragent) || !(request.useragent.startsWith("Amazon-Route53-Health-Check-Service"))"
    EOF
```

**Review by dhawton — CHANGES_REQUESTED** [review]
Also make sure you run `make gen` which will generate a test for the code snippet you added.

File: content/en/docs/tasks/observability/logs/telemetry-api/index.md [hunk 3]
```diff
@@ -132,6 +132,27 @@ $ kubectl apply -f @samples/open-telemetry/loki/otel.yaml@ -n istio-system
     EOF
     {{< /text >}}
 
+1. Filter health check access logs with CEL expression
+
+    The following configuration displays access logs only when the logs are not generated by the Amazon Route53 Health Check Service.
```
Comment (craigbox) [inline content/en/docs/tasks/observability/logs/telemetry-api/index.md:137]:
```suggestion
    The following configuration displays access logs only when the logs are not generated by the Amazon Route 53 Health Check Service.
```

That will fix the final lint error.

**Review by craigbox — CHANGES_REQUESTED** [review]

**Comment by ericvn** [comment]
It might be beneficial to add a call to your snippet in the test.sh to verify that the snippet is good. Verifying the behavior is correct would also be beneficial, but noting we don't have user-agent in the util at this point.

**Comment by whitneygriffith** [comment]
> It might be beneficial to add a call to your snippet in the test.sh to verify that the snippet is good. Verifying the behavior is correct would also be beneficial, but noting we don't have user-agent in the util at this point.

I am trying decide if it's worth implementing user-agent in the util. I know the snippet works based on user feedback here [#47533](https://github.com/istio/istio/issues/47533). If you think its worth it Ii will implement it.

**Comment by ericvn** [comment]
> @ericvn I am trying decide if it's worth implementing user-agent in the util. I know the snippet works based on user feedback here [#47533](https://github.com/istio/istio/issues/47533). If you think it's worth it Ii will implement it.

@whitneygriffith to get the change in quicker we can probably ignore the test for right now. I notice the earlier bullet was added without a test as well. We could open an issue for someone to add tests for the two bullets (and maybe other missed items if there are any).

**Review by ericvn — APPROVED** [review]

**Review by dhawton — APPROVED** [review]

**Review by craigbox — APPROVED** [review]

**Pull request merged**